### PR TITLE
Added support for headers

### PR DESCRIPTION
**Pull request by asBrettisay**
We love this gem, and started using it to set headers on our requests based on the swagger defined parameters.

Not sure if this is the best way to handle it, I would appreciate any feedback.

### CODE REVIEW

File: lib/svelte/rest_client.rb [hunk 1]
```diff
@@ -18,9 +18,10 @@ class << self
       #   an exception will be raised
       #
       # @return [Faraday::Response] http response from the service
-      def call(verb:, url:, params: {}, options: {})
+      def call(verb:, url:, params: {}, options: {}, headers: nil)
         connection.send verb, url, params do |request|
           request.options.timeout = options[:timeout] if options[:timeout]
+          request.headers = headers unless headers.nil?
```
Comment (AgentAntelope) [inline lib/svelte/rest_client.rb]: Personally I'd set headers to an empty hash so this isn't needed.

( [this](https://github.com/lostisland/faraday/blob/master/lib/faraday/request.rb#L40) ensures that the header type will be kept if you just always assign it.

File: lib/svelte/generic_operation.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ class << self
       # @param configuration [Configuration] Swagger API configuration
       # @param parameters [Hash] payload of the request, i.e. `{ petId: 1}`
       # @param options [Hash] request options, i.e. `{ timeout: 10 }`
-      def call(verb:, path:, configuration:, parameters:, options:)
+      def call(verb:, path:, configuration:, parameters:, options:, headers: nil)
```
Comment (brafales) [inline lib/svelte/generic_operation.rb]: Could you add the new `headers` parameter in the documentation comment above?

**Comment by jonklein** [comment]
Is there still interest in getting this merged in?  I'm in need of the same functionality and would be happy to take over and make the requested changes if needed.

cc @asBrettisay

**Comment by brafales** [comment]
@jonklein happy to merge it with the requested changes :)

**Comment by telekid** [comment]
I've opened up a PR based upon this one to see if we can get this merged in: https://github.com/notonthehighstreet/svelte/pull/20

**Comment by asBrettisay** [comment]
@telekid Sorry to ghost on everyone! If you want to take it on I'm all for it.

**Comment by telekid** [comment]
@asBrettisay no worries at all! You did all the work, I just did a little bit of merging.

**Comment by brafales** [comment]
Superseeded by https://github.com/notonthehighstreet/svelte/pull/20

**Pull request closed without merging**